whipper/common/encode.py

```diff
@@ -89,3 +90,68 @@ def _tag(self):
         w.save()
 
         self.stop()
+
+
+class EmbedPictureTask(task.Task):
+    description = 'Embed picture to FLAC'
+
+    def __init__(self, track_path, cover_art_path):
+        self.track_path = track_path
+        self.cover_art_path = cover_art_path
+
+    def start(self, runner):
+        task.Task.start(self, runner)
+        self.schedule(0.0, self._embed_picture)
+
+    def _make_flac_picture(self, cover_art_filename):
```

Comment:
To make the multivalue `--cover-art` option work as expected when the value is `embed` maybe we can add an argument to this method, like this:

```python
def _make_flac_picture(self, cover_art_filename, preserve=True):
        """
        Given a path to a jpg/png file, return a FLAC picture for embedding.

        The embedding will be performed using the mutagen module.

        :param cover_art_filename: path to cover art image file
        :type cover_art_filename: str
        :param preserve: whether to keep the downloaded cover
                         as standalone file or to delete it
        :type preserve: bool
        :returns: a valid FLAC picture for embedding
        :rtype: mutagen.flac.Picture or None
        """
````

Reply:
Shouldn't preserve be used somewhere else in `cd.py` ? Putting it to `_make_flac_picture` seems a little excessive to me. Maybe after looping through all the tracks then we should delete it.

Reply:
If we're sure to perform the file removal (`cover.jpg`) only after the task has been completed, then I agree with you.

Reply:
@ABCbum feel free to commit this, to remove the file a simple `os.remove(file_path)` should be enough. :wink: